driver/pace/driver/driver.py

```diff
@@ -407,7 +416,7 @@ def _critical_path_step_all(
                 )
                 if not self.config.disable_step_physics:
                     self._step_physics(timestep=dt)
-                self.end_of_step_actions(step)
+            self.end_of_step_actions(step)
```

Comment:
This looks suspect to me, so I'm calling it out. This used to be under the `mainloop` timer. Is it intended to move it out of that?

Reply:
I'm ok with this then as long as it's parallel to what we're measuring in the Fortran code. Is the IO part of the measurement there?